src/enums.ts

```diff
@@ -110,26 +129,32 @@ export function generateEnumToJson(ctx: Context, fullName: string, enumDesc: Enu
   const chunks: Code[] = [];
 
   const functionName = camelCase(fullName) + 'ToJSON';
-  chunks.push(
-    code`export function ${def(functionName)}(object: ${fullName}): ${
-      ctx.options.useNumericEnumForJson ? 'number' : 'string'
-    } {`
-  );
+  const paramType = options.enumUnspecifiedAsUndefined ? `${fullName} | undefined` : fullName;
+  const returnType = options.useNumericEnumForJson
+    ? 'number'
+    : 'string' + (options.enumUnspecifiedAsUndefined ? ' | undefined' : '');
+  chunks.push(code`export function ${def(functionName)}(object: ${paramType}): ${returnType} {`);
   chunks.push(code`switch (object) {`);
 
   for (const valueDesc of enumDesc.value) {
-    if (ctx.options.useNumericEnumForJson) {
+    if (options.useNumericEnumForJson) {
       chunks.push(code`case ${fullName}.${valueDesc.name}: return ${valueDesc.number};`);
-    } else {
-      chunks.push(code`case ${fullName}.${valueDesc.name}: return "${valueDesc.name}";`);
+      continue;
     }
+
+    if (options.enumUnspecifiedAsUndefined && valueDesc.number === 0) {
+      chunks.push(code`case undefined: return undefined;`);
```

Comment:
I.e. per my other comment, I think this should be `case undefined: return "${valueDesc.name}"`.